### PR TITLE
new case: nwfilter_binding_dumpxml test

### PR DESCRIPTION
**Pull request by juteman**
test virsh command nwfilter_binding_dumpxml

Signed-off-by: qiulzhan <qiulzhan@redhat.com>

### CODE REVIEW

File: libvirt/tests/cfg/nwfilter/nwfilter_binding_dumpxml.cfg [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    start_vm = "yes"
+    status_error = "no"
+    kill_vm = "yes"
+    option = "--type network --source default"
+    newfilter_1 = "clean-traffic"
+    newfilter_2 = "allow-dhcp-server"
+    alias_name = "net0"
+    source_network = "default"
+    variants:
+        - variable_notation:
```
Comment (chunfuwen) [inline libvirt/tests/cfg/nwfilter/nwfilter_binding_dumpxml.cfg]: variable_notation is not meaningful, maybe find one name,which is close to the feature

File: libvirt/tests/src/nwfilter/nwfilter_binding_dumpxml.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
+import logging
+import os
+
+from virttest import virsh
+from virttest import libvirt_xml
+from virttest import data_dir
```
Comment (chunfuwen) [inline libvirt/tests/src/nwfilter/nwfilter_binding_dumpxml.py]: add one blank line to separate imported package groups

File: libvirt/tests/src/nwfilter/nwfilter_binding_dumpxml.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+from avocado.utils import process
+
+
+def run(test, params, env):
+    """
+    Test virsh nwfilter-binding-list
+
+    1)Prepare parameters
+    2)Run nwfilter_binding_list command
+    3)check result
```
Comment (chunfuwen) [inline libvirt/tests/src/nwfilter/nwfilter_binding_dumpxml.py]: capitalize the first letter

File: libvirt/tests/src/nwfilter/nwfilter_binding_dumpxml.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    4)Clean env
+    """
+    vm_name = params.get("main_vm")
+    vm = env.get_vm(vm_name)
+    new_filter_1 = params.get("newfilter_1")
```
Comment (chunfuwen) [inline libvirt/tests/src/nwfilter/nwfilter_binding_dumpxml.py]: no meaningful variable name, 'new' is too general

File: libvirt/tests/src/nwfilter/nwfilter_binding_dumpxml.py [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    for i in range(len(params_key_2)):
+        params_dict = {}
+        params_dict['name'] = params[params_key_2[i]]
+        params_dict['value'] = params['dhcp_value_%s' % i]
+        filter_param_list_2.append(params_dict)
+    filterref_dict_2 = {}
+    filterref_dict_2['name'] = new_filter_2
+    filterref_dict_2['parameters'] = filter_param_list_2
+
+    def set_env():
```
Comment (chunfuwen) [inline libvirt/tests/src/nwfilter/nwfilter_binding_dumpxml.py]: set_env rename to create_netfilter ?

File: libvirt/tests/src/nwfilter/nwfilter_binding_dumpxml.py [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+        virsh.start(vm_name, debug=True)
+        # list binding port dev
+        ret = virsh.nwfilter_binding_list(debug=True)
+        utlv.check_exit_status(ret, status_error)
+        virsh.nwfilter_binding_dumpxml(new_iface_1.target['dev'],
+                                       to_file=vnet0_xml, debug=True)
+        virsh.nwfilter_binding_dumpxml(new_iface_2.target['dev'],
+                                       to_file=vnet1_xml, debug=True)
+        # check dump filterbinding can pass xml validate
+        vnet0_cmd = "virt-xml-validate %s" % vnet0_xml
```
Comment (chunfuwen) [inline libvirt/tests/src/nwfilter/nwfilter_binding_dumpxml.py]: check or validate function can be wrapped into one function

**Comment by kylazhang** [comment]
Merge the PR as he left and the update will be given later if needed.

**Pull request merged**